packages/target-ethers-v5/README.md

```diff
@@ -21,3 +21,49 @@
 This package requires TypeScript >= 3.8. If you need support for earlier TS versions check out: 1.0 version of this package.
 
 ## [TypeChain readme](https://github.com/ethereum-ts/TypeChain)
+
+## Contract typings
+
+The main files generated by this target are `<contract-name>.d.ts`. They declare typesafe interfaces for your contracts on top of ethers `Contract` instances:
+* typed contract's methods, available both at `contract.someMethod(...)` and `contract.functions.someMethod(...)`
+* typed events in `contract.interface.events.AnEvent` and filters in `contract.filters.AnEvent`
+* typed method gas estimates in `contract.estimate.someMethod`
+* overrides for the event listener methods (`on`, `once`, etc) that return the same contract type.
+
+Note: these are just _type declarations_ to help you call the blockchain properly, so they're not available at runtime, and all of the contracts are still instances of the same `Contract` class.
+
+## Contract factories
+
+This target also generates a concrete factory class for each contract, to help you deploy or connect to contract instances. The factory classes are an extension of ethers' `ContractFactory`. They serve two main purposes:
+* wrap passing contract ABI and bytecode to the `ContractFactory` class, so you don't have to load and parse the JSON manually
+* provide a correctly typed interface to `ContractFactory` (since it returns plain `Contract` instances).
+
+Abstract contracts or solidity interfaces are handled a bit different, because they have no bytecode. For those, a simplified factory is generated that doesn't extends `ContractFactory`, and only includes the static `connect` method, so you can easily connect to a deployed instance without having to pass the ABI manually.
+
+## Basic example
+
+Suppose you have an `Erc20Token.sol` solidity interface and a `DummyToken.sol` contract implementing it.
+
+```typescript
+import { BigNumber } from 'ethers/utils';
+import { Wallet } from 'ethers';
+
+import { DummyTokenFactory } from 'typechain-out-dir/DummyTokenFactory';
+import { DummyToken } from 'typechain-out-dir/DummyToken';
+import { Erc20TokenFactory } from 'typechain-out-dir/Erc20TokenFactory';
+
+const provider = getYourProvider(...);
+
+// use the concrete contract factory if you need to operate on the bytecode (ie. deploy)
+async function deployTestToken(ownerPK: string): Promise<DummyToken> {
+    const owner = new Wallet(ownerPK, provider);
+    return new DummyTokenFactory(owner).deploy();
+}
+
+// to call existing contracts, a factory for both the concrete contract and for the interface
+// can be used since the ABI is the same
+async function getTokenBalance(walletAddress: string, tokenAddress: string): Promise<BigNumber> {
+    const token = Erc20TokenFactory.connect(tokenAddress, provider);
+    return token.functions.balanceOf(walletAddress);
```

Comment:
The `functions` bucket, `token.functions.balanceOf` now returns `Promise<{0: BigNumber}>`. It is related to multiple outputs returned from smart contract. So it needed that `.functions.` to be not used for ordinary purpose
```suggestion
    return token.balanceOf(walletAddress);
```